apps/api/src/test-data/buildEdgeConnectionTestData/buildDualEdgeConnectionTestData/index.ts

```diff
@@ -1,9 +1,88 @@
-import { EdgeConnection } from '../../../domain/models/context/edge-connection.entity';
+import { EdgeConnectionContextType } from 'apps/api/src/domain/models/context/types/EdgeConnectionContextType';
+import { resourceTypes } from 'apps/api/src/domain/types/resourceTypes';
+import { InternalError } from 'apps/api/src/lib/errors/InternalError';
+import formatResourceCompositeIdentifier from 'apps/api/src/view-models/presentation/formatResourceCompositeIdentifier';
+import {
+    EdgeConnection,
+    EdgeConnectionMember,
+    EdgeConnectionMemberRole,
+    EdgeConnectionType,
+} from '../../../domain/models/context/edge-connection.entity';
+import buildOneSelfEdgeConnectionForEachResourceType from '../buildSelfConnectionTestData/buildOneSelfEdgeConnectionForEachResourceType';
 import buildOneDualEdgeConnectionForEveryContextType from './buildOneDualEdgeConnectionForEveryContextType';
 import buildOneFromConnectionForInstanceOfEachResourceType from './buildOneFromConnectionForInstanceOfEachResourceType';
 import buildOneToConnectionForInstanceOfEachResourceType from './buildOneToConnectionForInstanceOfEachResourceType';
 
+const buildDummyNoteForDualConnection = (
+    toMember: EdgeConnectionMember,
+    fromMember: EdgeConnectionMember
+): string =>
+    [
+        `That is why`,
+        formatResourceCompositeIdentifier(toMember.compositeIdentifier),
```

Comment:
Again, do we want to set apart dynamic vars in these messages?  Just worried that if they are for whatever reason missing or undefined we might miss it.

Reply:
It won't be missing or undefined. `validateTestData` ensures this data is good. Keep the context in mind. I am generating dummy data that should look realistic if you pull it through a view model to the frontend.